src/services/Process.php

```diff
@@ -473,6 +473,15 @@ public function processFeed($step, $feed, &$processedElementIds, $feedData = nul
 
         // Save the element
         if ($this->_service->save($element, $feed)) {
+
+            // save user's preferences only after user has been successfully saved
+            if (isset($attributeData['preferredLocale'])) {
```

Comment:
Should probably also check that it’s an actual user element here in the case where there’s a custom field named "preferredLocale" on any other element?

Reply:
I was going to say that this is only kicking in for attribute data, not custom fields data. Custom fields data is in the `$fieldData` array. And User element is the only one with the `preferredLocale` attribute, but actually, you could create your own custom element with that attribute, so yes, I added that check. Thanks!